[TensorExpr] Fix lowerings for aten::view and aten::reshape.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65555
* #65859
* __->__ #65852
* #65843
* #65554

Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

Commit:
[TensorExpr] Fix lowerings for aten::view and aten::reshape.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

Commit:
Update on "[TensorExpr] Fix lowerings for aten::view and aten::reshape."


Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

Commit:
Update on "[TensorExpr] Fix lowerings for aten::view and aten::reshape."


Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

Commit:
Update on "[TensorExpr] Fix lowerings for aten::view and aten::reshape."


Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

navahgar commented on torch/csrc/jit/tensorexpr/operators/misc.cpp:
```diff
@@ -382,12 +382,11 @@ Tensor computeExpand(
+    at::Device device) {
   auto A = c10::get<BufHandle>(inputs[0]);
   if (A.ndim() == 0) {
     return Compute(
```
This is not supporting `aten::view` in the general form. How are we restricting the cases we handle with `view`? 

Probably not directly relevant to this PR, but I just noticed this.

ZolotukhinM replied:
That's right, we're modeling `view` as `reshape`, which is not correct in a general case. We don't accept `view` in the fuser though, so this only is used in AOT scenario where users need to opt in.

Commit:
Update on "[TensorExpr] Fix lowerings for aten::view and aten::reshape."


Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

Commit:
Update on "[TensorExpr] Fix lowerings for aten::view and aten::reshape."


Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

Commit:
Update on "[TensorExpr] Fix lowerings for aten::view and aten::reshape."


Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

Commit:
Update on "[TensorExpr] Fix lowerings for aten::view and aten::reshape."


Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

Commit:
Update on "[TensorExpr] Fix lowerings for aten::view and aten::reshape."


Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).

Commit:
Update on "[TensorExpr] Fix lowerings for aten::view and aten::reshape."


Differential Revision: [D31286024](https://our.internmc.facebook.com/intern/diff/D31286024)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31286024).
